File: leetcode/easy/keep-multiplying-found-values-by-two.ts
// https://leetcode.com/problems/keep-multiplying-found-values-by-two/

// MINE: recursion
/*
function findFinalValue(nums: number[], o: number): number {
    if(!nums.includes(o)) return o
    return findFinalValue(nums, 2*o)
};
*/

// ANSWER: array
function findFinalValue(nums: number[], o: number): number {
  const A = new Array(1001).fill(0)
  for (let n of nums) ++A[n]
  while (o < 1001 && A[o]) o *= 2
  return o
}

File: leetcode/easy/keep-multiplying-found-values-by-two.py
# https://leetcode.com/problems/keep-multiplying-found-values-by-two/

class Solution:
    def findFinalValue(self, nums: List[int], o: int) -> int:
        A = [0]*1001
        for x in nums:
            A[x] += 1
        while o <= 1000 and A[o]:
            o *= 2
        return o
